### ISSUE TITLE
Error: timed out

### ISSUE THREAD

**Issue by Kyliux** [TRON-US/go-btfs#840]
btfs get <some hashabout 4 Go >

Error: timed out

...

**Comment by taiyangc** [comment]
Files can timeout if it's unreachable or the file is big and the timeout is too low.

**Comment by Kyliux** [comment]
It's the content of the public folder of a static Hugo website. I wanted to try website hosting on bittorrent. Seems like that's not gonna work..

**Comment by taiyangc** [comment]
Where did you upload it? You need to make sure it's on a host your local node can connect to. Or use https://gateway.btfs.io/

**Comment by Kyliux** [comment]
I uploaded it on btfs, here is the hash : QmZ3LrLqgof687oGmMwWJTHBfC7dMGqtUX1tJE2T3jXv76 I also payed a month for it so it doesn't need to stay on the local host.
you can try the gateway it won't work.

**Comment by taiyangc** [comment]
Not all hosts are reliable. We will try some testing to see if the host coverage can be improved.

**Comment by Kyliux** [comment]
Are you saying that this protocol does not offer by default enough redundancy to keep the data online in a decentralized way ?! I am glad I did not remove my data from my hard drive...

**Comment by taiyangc** [comment]
@Kyliux BTFS offers 3x redundancy by default. We are just evaluating whether this is enough given that hosts *can be* unreliable. Stay tuned on the results. It could be just *your* files.

**Comment by Kyliux** [comment]
I uploaded it again seems like it gives the same Hash see log : 

```
USERNAME$ btfs add --chunker=reed-solomon -r /Users/USERNAME/Downloads/testblog/blog/public 
added QmZ3LrLqgof687oGmMwWJTHBfC7dMGqtUX1tJE2T3jXv76 public
 3.82 GiB / 3.82 GiB [===================================================================================================] 100.00%
btfs storage upload QmZ3LrLqgof687oGmMwWJTHBfC7dMGqtUX1tJE2T3jXv76 --storage-length=35 --price=750000
{"ID":"51d21d93-7145-499b-ae88-50cfd5645516"}

--> [2020-12-18T15:06:43+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: submit, msg: Hosts found! Checking wallet balance and submitting contracts to escrow.
[2020-12-18T15:06:43+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: submit:check-balance-req-singed, msg: Hosts found! Checking wallet balance and submitting contracts to escrow.
[2020-12-18T15:06:44+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: submit:ledger-channel-commit-signed, msg: Hosts found! Checking wallet balance and submitting contracts to escrow.
[2020-12-18T15:06:45+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: pay, msg: Contracts submitted! Confirming the escrow payment.
[2020-12-18T15:06:45+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: pay:payin-req-signed, msg: Contracts submitted! Confirming the escrow payment.
[2020-12-18T15:06:47+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: guard, msg: Payment successful! Preparing meta-data and challenge questions.
[2020-12-18T15:06:47+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: guard:file-meta-signed, msg: Payment successful! Preparing meta-data and challenge questions.
[2020-12-18T15:06:59+01:00] session: 51d21d93-7145-499b-ae88-50cfd5645516 entered state: error, msg: GuardClient: rpc error: code = Unknown desc = Persistence manager: such file hash already exist for such renter
```
Since the hash is the same, i guess the upload was successfull just like the first time and **my files** are ok

**Comment by taiyangc** [comment]
You will get a refund when failing to upload the same file.

**Comment by taiyangc** [comment]
After investigating further, we discovered the availability of shards after ~1 week storage dropped significantly, bringing the probability of file still available to below 70% (according to a sample of 100 files we recently collected).

This can be due to many factors including but limited to 1) airdrop rewards ara no longer incentivizing 2) users abusing the old airdrop upload mechanism have withdrawn hosts 3) simply not enough incentive for hosting files 4) time zone / network region availability issues.

Those are the facts. We are working on bringing up the availability. Likely 1) better incentives to encourage long-term hosting 2) releasing decentralized repair feature to help keep recovering missing shards sooner and 3) other improvements to run the system more efficiently. There will be further updates on this soon.
